src/sha3.ts

```diff
@@ -10,19 +11,26 @@ import {
   HashXOF,
 } from './utils';
 
+const _0n = _n(0);
+const _1n = _n(1);
+const _2n = _n(2);
+const _7n = _n(7);
+const _256n = _n(256);
+const _0x71n = _n(0x71);
+
 // Various per round constants calculations
 const [SHA3_PI, SHA3_ROTL, _SHA3_IOTA]: [number[], number[], bigint[]] = [[], [], []];
-for (let round = 0, R = 1n, x = 1, y = 0; round < 24; round++) {
+for (let round = 0, R = _1n, x = 1, y = 0; round < 24; round++) {
   // Pi
   [x, y] = [y, (2 * x + 3 * y) % 5];
   SHA3_PI.push(2 * (5 * y + x));
   // Rotational
   SHA3_ROTL.push((((round + 1) * (round + 2)) / 2) % 64);
   // Iota
-  let t = 0n;
+  let t = _0n;
   for (let j = 0; j < 7; j++) {
-    R = ((R << 1n) ^ ((R >> 7n) * 0x71n)) % 256n;
-    if (R & 2n) t ^= 1n << ((1n << BigInt(j)) - 1n);
+    R = ((R << _1n) ^ ((R >> _7n) * _0x71n)) % _256n;
+    if (R & _2n) t ^= _1n << ((_1n << BigInt(j)) - _1n);
```

Comment:
```suggestion
    if (R & _2n) t ^= _1n << ((_1n << _n(j)) - _1n);
```

Also a top-level expectation of `BigInt` being available.